src/form/input/number/NumberInput.tsx

```diff
@@ -0,0 +1,46 @@
+import React from "react";
+
+import Input from "../Input";
+import {NumberInputProps} from "./util/numberInputTypes";
+import {
+  delocalizeNumberInputValue,
+  localizeNumberInputValue
+} from "./util/numberInputUtils";
+
+const NumberInput = React.forwardRef<HTMLInputElement, NumberInputProps>((props, ref) => {
+  const {formatProps = {}, maximumFractionDigits = 0, value, onChange, ...rest} = props;
```

Comment:
@jamcry  @yasincaliskan  I think we should add `pattern={"[0-9]*"}` attribute too. In this way, If the values contain non-digit characters, the element matches the `:invalid` CSS pseudo-classes.

Reply:
That sounds a good idea Anil thanks! 
However, we already have a utility which is `isNumberInputValueNotAValidNumber`. So, if enter an invalid character, it will be deleted actually.
We might change the approach if we want to use `:invalid` pseudo-class. (I think it can provide better UX) Or we can provide `pattern` prop optionally and remove `isNumberInputValueNotAValidNumber` check to be able to enter invalid characters. 
Let's discuss it at the next team meeting.

Reply:
I think for this PR, we can skip that because as you said we already have a custom validation.

We can create an issue for that, if we can migrate to native validation, it'd be the best, but we would need to test it very well